### DISCUSSION TITLE
Conditional filtering (advice request)

### DISCUSSION THREAD

**Discussion by yohplala** [vaexio/vaex#1984] [Q&A]
Hi,
I would like to filter a dataframe, depending conditions defined elsewhere in the code.
It seems I can do so by using filters defined as a string.
I have however not found this documented. Please, is this acceptable or should this be avoided?

Working example:
```python
import pandas as pd
import vaex

# Setup test data.
pdf = pd.DataFrame({'a':[-1, 0, 2, 4]})
vdf = vaex.from_pandas(pdf)
# Conditions defined 'elsewhere' to decide applying related filters or not.
apply_filter1 = True
apply_filter2 = True

# Build complete filter as a string    <= is this acceptable?
filters = ""
if apply_filter1:
    filters = "(a >=0)"
if apply_filter2:
    filter = "(a <= 2)"
    filters += " & "+filter if filters else filter

filters
Out[31]: '(a >=0) & (a <= 2)'

# Show time!
vdf = vdf[vdf[filters]]

vdf
Out[32]: 
  #    a
  0    0
  1    2
```

Thanks for your feedback!
Bests

**Comment by Hasham04** [comment]
Why not just do this:

```
# in the case both apply_filter1 and apply_filter2 are true
if apply_filter1 and apply_filter2:
   vdf=vdf[(vdf.a>=0)& (vdf.a<=2)]
# in the case apply_filter1 is true and apply_filter2 is not and vice versa:
else:
     vdf=vdf[vdf.a>=0] if apply_filter1 and not apply_filter2 else vdf[vdf.a <= 2]
```

**Comment by yohplala** [comment]
Well there could be other filters. The idea is to do this incrementally.
But as I understand vaex, re-reading the FAQ, I think it does not change perf to directly apply all filters at once or one by one (I should check).

So in the end, I am retaining simply:
```python
if apply_filter1:
    vdf = vdf[vdf['a'] >= 0]
if apply_filter2:
    vdf = vdf[vdf['a'] <= 2]
```

Thanks for your proposal!
Bests

**Comment by maartenbreddels** [comment]
yeah, performance shouldn't matter, because we don't eagery compute the filter.
